src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/ActiveConfiguredProjectsIgnoringTargetFrameworkProvider.cs

```diff
@@ -0,0 +1,180 @@
+        // Read/writes for both these fields must be done within a lock to keep them in sync.
+        private ImmutableDictionary<string, ConfiguredProject> _cachedConfiguredProjectsMap;
+        private string _cachedTargetFrameworks;
+
+        private IDisposable _evaluationSubscriptionLink;
+
+        [ImportingConstructor]
+        public ActiveConfiguredProjectsIgnoringTargetFrameworkProvider(
+            IUnconfiguredProjectServices services,
+            IUnconfiguredProjectCommonServices commonServices,
+            [Import(ExportContractNames.Scopes.UnconfiguredProject)]IProjectAsynchronousTasksService tasksService,
+            IActiveConfiguredProjectSubscriptionService activeConfiguredProjectSubscriptionService)
+            : base(commonServices.ThreadingService.JoinableTaskContext)
+        {
+            Requires.NotNull(services, nameof(services));
+            Requires.NotNull(commonServices, nameof(commonServices));
+            Requires.NotNull(tasksService, nameof(tasksService));
+            Requires.NotNull(activeConfiguredProjectSubscriptionService, nameof(activeConfiguredProjectSubscriptionService));
+
+            _services = services;
+            _commonServices = commonServices;
+            _tasksService = tasksService;
+            _activeConfiguredProjectSubscriptionService = activeConfiguredProjectSubscriptionService;
+        }
+
+        [ProjectAutoLoad(startAfter: ProjectLoadCheckpoint.ProjectFactoryCompleted)]
+        [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+        internal async Task OnProjectFactoryCompletedAsync()
+        {
+            await InitializeCoreAsync(CancellationToken.None).ConfigureAwait(false);
+        }
+
+        protected override async Task InitializeCoreAsync(CancellationToken cancellationToken)
+        {
+            using (_tasksService.LoadedProject())
+            {
+                await GetActiveConfiguredProjectsMapAsync().ConfigureAwait(false);
+
+                // Listen to changes to "TargetFrameworks" property.
+                var watchedEvaluationRules = Empty.OrdinalIgnoreCaseStringSet.Add(ConfigurationGeneral.SchemaName);
+                _evaluationSubscriptionLink = _activeConfiguredProjectSubscriptionService.ProjectRuleSource.SourceBlock.LinkTo(
+                        new ActionBlock<IProjectVersionedValue<IProjectSubscriptionUpdate>>(OnProjectChanged),
+                        ruleNames: watchedEvaluationRules, suppressVersionOnlyUpdates: true);
+            }
+        }
+
+        private async Task OnProjectChanged(IProjectVersionedValue<IProjectSubscriptionUpdate> e)
+        {
+            IProjectChangeDescription projectChange = e.Value.ProjectChanges[ConfigurationGeneral.SchemaName];
+            if (projectChange.Difference.ChangedProperties.Contains(ConfigurationGeneral.TargetFrameworksProperty))
+            {
+                var targetFrameworks = projectChange.After.Properties[ConfigurationGeneral.TargetFrameworksProperty];
```

Comment:
I need convicing that consumers of this class want the TargetFrameworks in the map. These are user written forms, are not canonical (there can be many representations of the same target framework (say net45 vs net4.5) and shouldn't bleed to outside consumers. I want TargetFrameworkMoniker of each configured project to be exchange form; it has a single representation and is well known throughout VS APIs.